src/commands/modping.ts

```diff
@@ -0,0 +1,160 @@
+import { SlashCommandBuilder } from '@discordjs/builders';
+import { getDB } from '@/db';
+import type { ChatInputCommandInteraction, GuildMember } from 'discord.js';
+import { ModPingSettings } from '@/models/modPingSettings';
+
+async function interactionHandler(interaction: ChatInputCommandInteraction): Promise<void> {
+    if (interaction.commandName !== 'mod-ping') {
+        return;
+    }
+
+    const roleId = getDB().get('roles.mod-ping');
+    if (!roleId) {
+        await interaction.reply({ content: `Missing mod-ping role ID!`, ephemeral: true });
+        return;
+    }
+
+    let message = 'An error occurred while processing your request.';
+
+    try {
+        const subcommandGroup = interaction.options.getSubcommandGroup(false);
+        const subcommand = interaction.options.getSubcommand();
+
+        if (subcommand === 'toggle') {
```

Comment:
The different subcomands could be moved to separate functions. It would make this easier to read with less complex nesting.

Reply:
ill separate them